tests/check_xcode.sh

```diff
@@ -8,72 +8,98 @@
 # mod-pbxproj should already have been installed (through PIP) before
 # running this script.
 
+has_file() {
+	fgrep -q "path = source/${1};" ${XPROJECT}/project.pbxproj
+}
 
-# Determine path of the current script
-HERE=$(cd `dirname $0` && pwd)
+has_encoded_file() {
+	fgrep "path = source/${1};" ${XPROJECT}/project.pbxproj | fgrep -q "fileEncoding = 4;"
+}
+
+add_new() {
+	local OPTS='--tree <group> --parent source'
+	echo -n "'${1}': "
+	python3 -m pbxproj file "${XPROJECT}" "source/${1}" ${OPTS} --target EndlessSky ${2}
+	# Make sure the file was added with Unicode encoding (fileEncoding = 4;)
+	if ! has_encoded_file ${1}; then
+		local GOOD_STR="isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c"
+		local BAD_STR="isa = PBXFileReference; lastKnownFileType = sourcecode.c"
+		local BACKUP=${XPROJECT}/project.pbxproj.bak
+		sed -i.bak s/"${BAD_STR}"/"${GOOD_STR}"/ ${XPROJECT}/project.pbxproj
+		if [ $? -ne 0 ]; then
+			echo "Using sed to enforce unicode encoding failed! Restoring backup XCode project"
+			cp "${BACKUP}" ${XPROJECT}/project.pbxproj
+		fi
+		rm -f "${BACKUP}"
+	fi
+}
+
+move_existing() {
+	# The simplest method to "move" a file in XCode is to update the referred file path. These lines look like
+	# A96862DB1AE6FD0A004FE1FE /* BankPanel.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; name = BankPanel.cpp; path = source/BankPanel.cpp; sourceTree = "<group>"; };
+	local basename=$(basename ${1})
+	local BACKUP=${XPROJECT}/project.pbxproj.bak
+	# Since the normal sed delimiter / is used in file paths, we will use _ as the delimiter:
+	sed -i.bak "\_path = source/${basename};_ s_/${basename}_/${1}_" ${XPROJECT}/project.pbxproj
```

Comment:
It will only match the path, because the sed delimiter is `_` and not `/`. So the target is `/BankPanel.*`, and the replacement is `/panels/BankPanel.*`

To support subfolder swaps, this will be updated anyway and not be as easily overlooked

Reply:
You're right, the slash indeed ensures that only the path is matched.